[DO NOT MERGE][DPE-6480] Add docs landing page first draft (#57)
canonical/kyuubi-k8s-operator: docs/overview.md

# Charmed Apache Kyuubi K8s

> Charmed Apache Kyuubi K8s, an open-source [Kubernetes charm](https://juju.is/docs/olm/charmed-operator) for Apache Kyuubi, is a distributed and multi-tenant gateway to provide serverless SQL on Data Warehouses and Lakehouses built with the Charmed Apache Spark solution.

Reviewer: ```suggestion
Charmed Apache Kyuubi K8s, an open-source [Kubernetes charm](https://juju.is/docs/olm/charmed-operator) for Apache Kyuubi, is a distributed and multi-tenant gateway to provide serverless SQL capabilities on Data Warehouses and Lakehouses leveraging on Apache Spark computational engines. The Charmed Apache Kyuubi K8s operator is integrated in the Charmed Apache Spark solution.
```

> Charmed Apache Kyuubi K8s provides users with a JDBC-compliant endpoint which can be used by JDBC-compliant tools and analytics platforms to execute SQL queries on the Charmed Spark cluster.

Reviewer 2: Question 2 - what the product does?
**Suggestion**:
```suggestion
Charmed Apache Kyuubi K8s provides a JDBC endpoint which can be used by JDBC-compliant tools to execute SQL queries on a Charmed Apache Spark cluster. 
```
That seems like the Data Warehouse part of the previous description. Can we add something specific for the second (Lakehouse) part?

> It is part of an open source, end-to-end, production-ready data platform on top of cloud native technologies provided by Canonical.

Reviewer 2: I'm not sure whether we need this part but if we do need it, consider adding some link(s) to it. Maybe [Spark](https://canonical.com/data/docs/spark/k8s/t-overview) or at least https://canonical.com/data

Author: Added in f134856e2cf10061a8035d776c34210c0ffedd00. I think the initial meaning of this part was to emphasize that this charm is not self-sufficient

> 

Reviewer 2: Question 3 - what need the product meets?
```suggestion

Using Charmed Apache Kyuubi K8s minimizes the barriers and costs for end-users to use Charmed Apache Spark at the client side. At the server-side, multi-tenant architecture of Apache Kyuubi provides the administrators a way to achieve computing resource isolation, data security, high availability, high client concurrency, etc.
```

> <!--

Reviewer 2: Question 4 - whom the product is useful for?
```suggestion

Charmed Apache Kyuubi K8s is useful for administrators of Charmed Apache Spark solutions who want to provide serverless SQL query capability or multi-tenancy for their end-users.

<!--
```

Author: Ah, great; I did not realize I missed the fourth question, thank you so much

# Navigation

SEE TEMPLATE
-->

## Usage

<!--TODO: Remove this entire section once we have a dedicated page-->

To deploy this charm, use:

```shell
juju add-model <my-model>
juju deploy s3-integrator --channel latest/edge
juju deploy spark-integration-hub-k8s --channel edge
juju relate spark-integration-hub-k8s s3-integrator
juju deploy kyuubi-k8s --channel edge
juju relate spark-integration-hub-k8s kyuubi-k8s
```

> [note type=information]
> You may use a different object storage, such as `azure-storage-integrator`.
> [/note]

Reviewer 2: The note blocks aren't working on Discourse now:
```suggestion
> **Note**: You may use a different object storage, such as `azure-storage-integrator`.
```

The Integration Hub charm will take care of adding relevant configuration to the
Charmed Spark properties.
Get the JDBC endpoint with the following:

```shell
juju run kyuubi-k8s/leader get-jdbc-endpoint
```

> 

Reviewer: **suggestion** should we provide also some code to show how to use this with beeline?

Author: Added in 7a1eb3e7b44815f56886316d27a07924f07465b6. Do you think this is enough?
